packages/react/src/utils/use-module-query-client.ts

```diff
@@ -0,0 +1,147 @@
+import { CosmWasmClient } from '@cosmjs/cosmwasm-stargate'
+import * as React from 'react'
+
+import {
+  ABSTRACT_API_URL,
+  AbstractAccountId,
+  AbstractQueryClient,
+  CHAIN_DEPLOYMENT_QUERY,
+  graphqlRequest,
+} from '@abstract-money/core'
+import { UseQueryOptions, useQuery } from '@tanstack/react-query'
+import { useAccountId, useConfig } from '../contexts'
+
+interface ModuleQueryClientConstructor {
+  new (args: {
+    abstractQueryClient: AbstractQueryClient
+    accountId: AbstractAccountId
+    managerAddress: string
+    proxyAddress: string
+    moduleId: string
+  }): any
+}
+
+async function getModuleQueryClient<
+  TModule extends ModuleQueryClientConstructor,
+>({
+  client,
+  chain,
+  overrideApiUrl = ABSTRACT_API_URL,
+  accountId,
+  moduleId,
+  Module,
+}: {
+  client: CosmWasmClient
+  chain: string
+  overrideApiUrl?: string
+  accountId: AbstractAccountId
+  moduleId: string
+  Module: TModule
+}) {
+  // TODO: re-check if grabbing the first chain of the list is a good solution
+  const data = await graphqlRequest(overrideApiUrl, CHAIN_DEPLOYMENT_QUERY, {
+    chain,
+  })
```

Comment:
It's not a good solution - we'll need to have the chainName passed in as a parameter or provided in some sort of context

Reply:
Noted at https://linear.app/abstract-sdk/issue/APP-248/fix-the-chain-name-assertion-in-pseudogenerated-modules